internal/server/v1/warden/service.go

```diff
@@ -0,0 +1,175 @@
+package warden
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	shieldv1beta1rpc "buf.build/gen/go/gotocompany/proton/grpc/go/gotocompany/shield/v1beta1/shieldv1beta1grpc"
+	shieldv1beta1 "buf.build/gen/go/gotocompany/proton/protocolbuffers/go/gotocompany/shield/v1beta1"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/goto/dex/internal/server/reqctx"
+)
+
+//go:generate mockery --with-expecter --keeptree --case snake --name Doer
+
+const (
+	baseURL = "https://go-cloud.golabs.io"
+)
+
+type Service struct {
+	shieldClient shieldv1beta1rpc.ShieldServiceClient
+	doer         Doer
+}
+
+type Doer interface {
```

Comment:
instead of calling this doer, let's just call it `HttpClient` to be more explicit.

Reply:
We need a doer, so that we can mimic http call in test.

Reply:
yes, we need a interface, but let's call it HttpClient instead.

`warden.Doer` does not make sense.